process/coordinator/process.go

```diff
@@ -396,8 +396,8 @@ func (tc *transactionCoordinator) CreateMbsAndProcessCrossShardTransactionsDstMe
 	nrTxAdded := uint32(0)
 	nrMiniBlocksProcessed := 0
 
-	if hdr == nil || hdr.IsInterfaceNil() {
-		return miniBlocks, nrTxAdded, true
+	if check.IfNil(hdr) {
+		return miniBlocks, nrTxAdded, false
```

Comment:
If we return true, which is interpreted as hdrProcessedFinished in CreateMbsAndProcessCrossShardTransactionsDstMe (metaProcessor/shardProcessor), the loop would not be stopped and this will cause some problems like:

In shardProcessor:
Will set lastMetaHdr = currMetaHdr and continue the loop

In metaProcessor:
Will do this and continue the loop: 
			mp.hdrsForCurrBlock.hdrHashAndInfo[string(orderedHdrsHashes[i])] = &hdrInfo{hdr: currShardHdr, usedInBlock: true}
			hdrsAdded++
			hdrsAddedForShard[currShardHdr.GetShardID()]++

			lastShardHdr[currShardHdr.GetShardID()] = currShardHdr